pkg/runtime/optimize/hints.go

```diff
@@ -100,3 +103,21 @@ func (h *Direct) exec(tableName ast.TableName, rule *rule.Rule) (hintTables rule
 	hintTables[db0] = []string{tableName.String()}
 	return hintTables, nil
 }
+
+type CustomRoute struct {
+}
+
+func (c *CustomRoute) exec(tableName ast.TableName, r *rule.Rule, hints []*hint.Hint) (hintTables rule.DatabaseTables, err error) {
+	hintTables = make(map[string][]string)
+	for _, h := range hints {
```

Comment:
hints should be a collection of hints that contain only the router

Reply:
@binbin0325  what's the meaning of this?

Reply:
only routes for custom hints are processed